src/libAtomVM/nifs.c

```diff
@@ -4485,14 +4485,15 @@ static void *nif_code_all_available_fold(void *accum, const void *section_ptr, u
 
     struct CodeAllAvailableAcc *acc = (struct CodeAllAvailableAcc *) accum;
     size_t section_name_len = strlen(section_name);
-    if (section_name_len < 260) {
-        if (memcmp(".beam", section_name + section_name_len - 5, 5) == 0) {
+    if (LIKELY(section_name_len > 5 && section_name_len < 260)) {
+        size_t module_name_len = section_name_len - 5;
+        if (memcmp(".beam", section_name + module_name_len, 5) == 0) {
```

Comment:
why not using BEAM_CODE_FLAG? I think using it would make code simpler.

Reply:
I added a test for `BEAM_CODE_FLAG`. However I need the module name here, so I kept the logic to check for and remove the .beam suffix.